### ISSUE TITLE
No intellisense in bracketless if statement

### ISSUE THREAD

**Issue by mavericken** [Microsoft/nodejstools#340]
I can't get intellisense in if statement bodies that do not use brackets.  This is pretty minor and has an easy work around, but I think it would be a better experience if this was fixed.

Here is my version information:
node.js tools 1.0.30109.03
vs2013 12.0.31101.00 update 4

![image](https://cloud.githubusercontent.com/assets/7833752/9016275/c3c843f6-379d-11e5-88f1-02e535c2d248.png)

![image](https://cloud.githubusercontent.com/assets/7833752/9016283/deab9966-379d-11e5-983c-8c67992b6e10.png)

    var cat = "meow";
    if (true) cat;
    if (true) { cat };

**Comment by hoanhtien** [comment]
This can be fixed by removing `)` from [NormalCompletionAnalysis.cs#L99](https://github.com/Microsoft/nodejstools/blob/35817ffdaa1797f95628a22f588fc3ed885f884b/Nodejs/Product/Nodejs/Intellisense/NormalCompletionAnalysis.cs#L99). Not sure if it is the right fix though.
```C#
private string FixupCompletionText(string exprText) {
    if (exprText.EndsWith(".")) {
        exprText = exprText.Substring(0, exprText.Length - 1);
        if (exprText.Length == 0) {
            // don't return all available members on empty dot.
            return null;
        }
    } else {
        int cut = exprText.LastIndexOfAny(new[] { '.', ']', ')' }); // Remove ) from this list
        if (cut != -1) {
            exprText = exprText.Substring(0, cut);
        } else {
            exprText = String.Empty;
        }
    }
    return exprText;
} 
```

**Comment by dezsiszabi** [comment]
I tried removing the ')' from that array, but it didn't fix the issue for me.

Not sure, but I think the error is in this method in ReverseExpressionParser.cs:

```cs
public SnapshotSpan? GetExpressionRange(int nesting, out int paramIndex, out SnapshotPoint? sigStart, out string lastKeywordArg, out bool isParameterName, bool forCompletion = true, bool forSignatureHelp = false)
```

That returns the following `SnapshotSpan`:
`{?_v1_[24..34)_'(true) cat'}`

Shouldn't this be the "cat" part only?

Also maybe I misunderstand something (excuse me, I just cloned the repository yesterday), but I noticed that in the above mentioned `GetExpressionRange` we have a variable called `lastTokenWasKeywordArgAssignment` which is initially set to `false`.

Searching for every usage of this variable shows that it's used in seven assignments and in one if condition. The interesting thing is that all the assignments assign the value `false` to it.

The if condition looks like the following:

```cs
if (lastTokenWasKeywordArgAssignment &&
    token.ClassificationType.IsOfType(PredefinedClassificationTypeNames.Identifier) &&
    lastKeywordArg == null) { ... }
```

How can this ever be true?
